### DISCUSSION TITLE
Update of change in credentials on onedrive account

### DISCUSSION THREAD

**Discussion by siddharth-deshmukh** [abraunegg/onedrive#1545] [Q&A] [answered]
Hi, 
I installed onedrive on Ubuntu 20 following your instructions and everything worked for me. However, in onedrive authentication along with password my university also has another step of entering a numeric code received over the SMS. This SMS based authentication step is not every time but almost once in a month. So, I need support for updating these credentials on my local installation as now onedrive --monitor is giving an error. I believe this will also happen if I need to change the password of my one drive account. I am using this version: onedrive v2.4.12-6-g16fdd92.

Thanks in advance for your help.

Regards,
Siddharth

**Comment by abraunegg** [comment, marked as answer]
@siddharth-github 
> So, I need support for updating these credentials on my local installation as now onedrive --monitor is giving an error.


It would be useful for you to provide an example of the error you are receiving.

In all password or credential issues, the way to resolve this is to issue the following:
```
onedrive --logout
```

This will prompt you to re-authenticate the client.

**Reply by lennonab** [reply]
I seem to be having a similar issue (using Ubuntu package - ) with our university's recently implemented two-factor authentication (which normally sends a request to an app on my phone to approve logins (with an option for persistent approval of 90 days usually).

I've logged out of onedrive as recommended in the previous answer, which triggers the authorization procedure, but when I login to the provided address on my browser and paste the response uri back to the console I get a "HTTP 400 - Bad Request" and the following error log:
```
OneDrive returned a 'HTTP 400 - Bad Request' - gracefully handling error
onedrive.OneDriveException@src/onedrive.d(874): HTTP request returned status code 400 (Bad Request)
{
    "correlation_id": "076d7cb8-a70a-4309-808b-690a4a01f630",
    "error": "invalid_grant",
    "error_codes": [
        9002313
    ],
    "error_description": "AADSTS9002313: Invalid request. Request is malformed or invalid.\r\nTrace ID: cc44b353-4faa-4554-95b9-fc766e0f8100\r\nCorrelation ID: 076d7cb8-a70a-4309-808b-690a4a01f630\r\nTimestamp: 2021-09-02 12:25:06Z",
    "error_uri": "https:\/\/login.microsoftonline.com\/error?code=9002313",
    "timestamp": "2021-09-02 12:25:06Z",
    "trace_id": "cc44b353-4faa-4554-95b9-fc766e0f8100"
}
----------------
??:? [0x5649c7c2e9d9]
??:? [0x5649c7c2dcb5]
??:? [0x5649c7c2eb65]
??:? [0x5649c7c2cf58]
??:? [0x5649c7c3a7a8]
??:? void rt.dmain2._d_run_main2(char[][], ulong, extern (C) int function(char[][])*).runAll() [0x7efe046179db]
??:? _d_run_main2 [0x7efe046177ee]
??:? _d_run_main [0x7efe0461765d]
??:? __libc_start_main [0x7efe042030b2]
??:? [0x5649c7c055ed]`
```
Any help is much appreciated,
Alex

**Reply by lennonab** [reply]
Sorry, forgot to state the version I'm using: v2.3.13-1build2

**Reply by abraunegg** [reply]
@lennonab 
> Sorry, forgot to state the version I'm using: v2.3.13-1build2

You are using an unsupported client version. Please read https://github.com/abraunegg/onedrive/blob/master/docs/INSTALL.md *very* carefully.

**Reply by lennonab** [reply]
Only in the office occasionally at the moment so just getting a chance to try this out today again.  I've switched to the OpenSuse repository as per instructions and it seems to have authenticated now and is currently doing a "dry-run" so hopefully all good now.

Many thanks for the help and apologies for missing the docs on this issue.
